lib/resty/upstream/healthcheck.lua

```diff
@@ -145,6 +145,10 @@ local function peer_fail(ctx, is_backup, id, peer)
                 " failure(s)")
         peer.down = true
         set_peer_down_globally(ctx, is_backup, id, true)
+
+        if (ctx.hook_down) then
```

Comment:
Maybe we should respect the lua-resty codestyle here and drop the parenthesis? The condition is quite simple to grasp as well.